src/scandipwa/app/plugin/GTMHocWrapper.plugin.js

```diff
@@ -8,40 +8,31 @@ import {
 export const CONFIRM_ACCOUNT = 'confirm_account';
 
 import withGTM from '../component/GoogleTagManager/withGTM.hoc';
 
-// TODO forgot password without HOC
-const homePage = Class => withGTM(Class, HOME_PAGE);
-const categoryPage = Class => withGTM(Class, CATEGORY);
-const searchPage = Class => withGTM(Class, SEARCH);
-const productPage = Class => withGTM(Class, PDP);
-const cmsPage = Class => withGTM(Class, CMS_PAGE);
-const cartPage = Class => withGTM(Class, CART);
-const checkout = Class => withGTM(Class, CHECKOUT);
-const passwordChangePage = Class => withGTM(Class, PASSWORD_CHANGE);
-const confirmAccountPage = Class => withGTM(Class, CONFIRM_ACCOUNT);
-const myAccount = Class => withGTM(Class, CUSTOMER_ACCOUNT);
-const menuPage = Class => withGTM(Class, MENU);
-const urlRewrites = Class => withGTM(Class, URL_REWRITE);
+const wrapperFunction = (route) => (args, callback, instance) => {
+    const originalRender = callback.apply(instance, args);
+
+    return withGTM(originalRender, route);
+}
 
 export default {
-    'Route/HomePage/Container': { 'class': homePage },
-    'Route/CategoryPage/Container': { 'class': categoryPage },
-    'Route/SearchPage/Container': { 'class': searchPage },
-    'Route/ProductPage/Container': { 'class': productPage },
-    'Route/CmsPage/Container': { 'class': cmsPage },
-    'Route/CartPage/Container': { 'class': cartPage },
-    'Route/Checkout/Container': { 'class': checkout },
-    'Route/PasswordChangePage/Container': { 'class': passwordChangePage },
-    'Route/ConfirmAccountPage/Container': { 'class': confirmAccountPage },
-    'Route/MyAccount/Container': { 'class': myAccount },
-    'Route/MenuPage/Container': { 'class': menuPage },
-    'Route/UrlRewrites/Container': { 'class': urlRewrites },
+    'Route/HomePage/Container': { 'member-function': { render: wrapperFunction(HOME_PAGE) } },
+    'Route/CategoryPage/Container': { 'member-function': { render: wrapperFunction(CATEGORY) } },
+    'Route/ProductPage/Container': { 'member-function': { render: wrapperFunction(PDP) } },
+    'Route/SearchPage/Container': { 'member-function': { render: wrapperFunction(SEARCH) } },
+    'Route/CmsPage/Container': { 'member-function': { render: wrapperFunction(CMS_PAGE) } },
+    'Route/CartPage/Container': { 'member-function': { render: wrapperFunction(CART) } },
+    'Route/Checkout/Container': { 'member-function': { render: wrapperFunction(CHECKOUT) } },
+    'Route/PasswordChangePage/Container': { 'member-function': { render: wrapperFunction(PASSWORD_CHANGE) } },
+    'Route/ConfirmAccountPage/Container': { 'member-function': { render: wrapperFunction(CONFIRM_ACCOUNT) } },
+    'Route/MyAccount/Container': { 'member-function': { render: wrapperFunction(CUSTOMER_ACCOUNT) } },
+    'Route/MenuPage/Container': { 'member-function': { render: wrapperFunction(MENU) } },
+    'Route/UrlRewrites/Container': { 'member-function': { render: wrapperFunction(URL_REWRITE) } }
```

Comment:
This reduces the logic complexity a bit, but I liked the old option as well :D

Reply:
On our project we have noticed some issues related to old solution (that we found together with you) - some checkout things were not working, so one of my team member updated this logic, but then we noticed that plugins are not triggering at all.

This is his final solution that works for sure

Reply:
alright.